service/controller/aws/v14/resource/kubeconfig/desired.go

```diff
@@ -0,0 +1,74 @@
+package kubeconfig
+
+import (
+	"context"
+
+	"github.com/giantswarm/kubeconfig"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/client/k8srestconfig"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/cluster-operator/pkg/label"
+	"github.com/giantswarm/cluster-operator/pkg/v14/key"
+	awskey "github.com/giantswarm/cluster-operator/service/controller/aws/v14/key"
+)
+
+func (r *StateGetter) GetDesiredState(ctx context.Context, obj interface{}) ([]*corev1.Secret, error) {
+	cr, err := awskey.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	clusterGuestConfig := awskey.ClusterGuestConfig(cr)
+	apiDomain, err := key.APIDomain(clusterGuestConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	appOperator, err := r.certsSearcher.SearchAppOperator(clusterGuestConfig.ID)
```

Comment:
The cert secret will not exist until the certconfig CR has been processed by cert-operator. 

So we need to handle the timeout error and cancel the reconciliation. We will retry in the next controller loop.

See this example.

https://github.com/giantswarm/cluster-operator/blob/a186b43af9a728d321082d92347b8a8bc336ea38/pkg/v14/resource/namespace/current.go#L35